src/main/java/teammates/ui/template/AdminActivityLogTableRow.java

```diff
@@ -0,0 +1,211 @@
+    }
+    
+    // --------------- Additional generated fields ---------------
+    
+    public String getUserHomeLink() {
+        switch (activityLog.getRoleWithoutMasquerade()) {
+        case Const.ActivityLog.ROLE_STUDENT:
+            return Url.addParamToUrl(Const.ActionURIs.STUDENT_HOME_PAGE,
+                    Const.ParamsNames.USER_ID, activityLog.getGoogleId());
+        case Const.ActivityLog.ROLE_INSTRUCTOR:
+            return Url.addParamToUrl(Const.ActionURIs.INSTRUCTOR_HOME_PAGE,
+                    Const.ParamsNames.USER_ID, activityLog.getGoogleId());
+        default:
+            return null;
+        }
+    }
+    
+    public boolean getHasUserHomeLink() {
+        return activityLog.getRoleWithoutMasquerade().contains(Const.ActivityLog.ROLE_STUDENT)
+                || activityLog.getRoleWithoutMasquerade().contains(Const.ActivityLog.ROLE_INSTRUCTOR);
+    }
+    
+    public String getUserIdentity() {
+        String googleId = activityLog.getGoogleId();
+        if (!googleId.contentEquals(Const.ActivityLog.AUTH_NOT_LOGIN)
+                && !googleId.contentEquals(Const.ActivityLog.UNKNOWN)) {
+            return googleId;
+        }
+        
+        String email = activityLog.getEmail();
+        if (email != null && !email.contentEquals(Const.ActivityLog.UNKNOWN)) {
+            return email;
+        }
+        
+        String name = activityLog.getName();
+        if (name != null && !name.contentEquals(Const.ActivityLog.UNKNOWN)) {
+            return name;
+        }
+        return "";
+    }
+    
+    // --------------- 'is' fields to determine css class ---------------
+    
+    public boolean getIsUserAdmin() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_ADMIN);
+    }
+    
+    public boolean getIsUserInstructor() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_INSTRUCTOR);
+    }
+    
+    public boolean getIsUserStudent() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_STUDENT);
+    }
+    
+    public boolean getIsUserAuto() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_AUTO);
+    }
+    
+    public boolean getIsUserUnregistered() {
+        return activityLog.getRole().contains(Const.ActivityLog.ROLE_UNREGISTERED);
+    }
+    
+    public boolean getIsTimeTakenWarning() {
+        return activityLog.getTimeTaken() >= TIME_TAKEN_WARNING_LOWER_RANGE
+                && activityLog.getTimeTaken() <= TIME_TAKEN_WARNING_UPPER_RANGE;
+    }
+    
+    public boolean getIsTimeTakenDanger() {
+        return activityLog.getTimeTaken() > TIME_TAKEN_WARNING_UPPER_RANGE
+                && activityLog.getTimeTaken() <= TIME_TAKEN_DANGER_UPPER_RANGE;
+        // TODO : check why there is upper range
+    }
+    
+    public boolean getIsActionWarning() {
+        return activityLog.getAction().contains(Const.ACTION_RESULT_FAILURE);
+    }
+    
+    public boolean getIsActionDanger() {
```

Comment:
Again, do not mention `danger` and `warning`. The methods should be named as what they do: `isLoggingActionErrorReport`, `isLoggingActionFailure`.